src/consensus/upgrades.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2018 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "consensus/upgrades.h"
+
+const struct NUInfo NetworkUpgradeInfo[Consensus::MAX_NETWORK_UPGRADES] = {
+    {
+        /*.nBranchId =*/ 0,
+        /*.strName =*/ "Sprout",
+        /*.strInfo =*/ "The Zcash network at launch",
+    },
+    {
+        /*.nBranchId =*/ 0x74736554,
+        /*.strName =*/ "Test dummy",
+        /*.strInfo =*/ "Test dummy info",
+    },
+    {
+        /*.nBranchId =*/ 0x5ba81b19,
+        /*.strName =*/ "Overwinter",
+        /*.strInfo =*/ "TBD",
+    }
+};
+
+UpgradeState NetworkUpgradeState(
+    int nHeight,
+    const Consensus::Params& params,
+    Consensus::UpgradeIndex idx)
+{
+    assert(nHeight >= 0);
+    assert(idx >= Consensus::UPGRADE_SPROUT && idx < Consensus::MAX_NETWORK_UPGRADES);
+    auto nActivationHeight = params.vUpgrades[idx].nActivationHeight;
+
+    if (nActivationHeight == Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT) {
+        return UPGRADE_DISABLED;
+    } else if (nHeight >= nActivationHeight) {
+        // From ZIP ???:
+        //
+        // ACTIVATION_HEIGHT
+        //     The block height at which the network upgrade rules will come into effect.
+        //
+        //     For removal of ambiguity, the block at height ACTIVATION_HEIGHT - 1 is
```

Comment:
So there will be no tx embargo period?

Reply:
Not at the consensus layer, no. @zookozcash was only willing to consider it for Overwinter itself if necessary, not for the general mechanism (which is what this PR implements).